src/osd/ReplicatedPG.cc

```diff
@@ -5977,6 +5979,23 @@ int ReplicatedPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
       }
       break;
 
+    case CEPH_OSD_OP_ASSERT_INTERVAL:
+      ++ctx->num_read;
+      {
+	epoch_t epoch = op.assert_interval.epoch;
+	epoch_t pg_epoch = get_epoch();
```

Comment:
The assert_interval.epoch is coming from user mode and wouldn't be checked by any other code.  Below I check to make sure it is even a reasonable value and return ERANGE if it is way our of range.